applications/HDF5Application/custom_io/hdf5_file.h

```diff
@@ -85,216 +99,409 @@ class KRATOS_API(HDF5_APPLICATION) File
+     * @return std::vector<std::string>     List of link names.
+     */
     std::vector<std::string> GetLinkNames(const std::string& rGroupPath) const;
 
+    /**
+     * @brief Get the sub group names under a group.
+     *
+     * @param rGroupPath                    Group path.
+     * @return std::vector<std::string>     List of sub group names.
+     */
     std::vector<std::string> GetGroupNames(const std::string& rGroupPath) const;
 
+    /**
+     * @brief Get the data set names under a group.
+     *
+     * @param rGroupPath                    Group path.
+     * @return std::vector<std::string>     List of sub data set names.
+     */
     std::vector<std::string> GetDataSetNames(const std::string& rGroupPath) const;
 
+    /**
+     * @brief Add group to the path recursively.
+     *
+     * This method creates all the parent groups to create the final @ref rPath.
+     *
+     * @throws If any of the parents in the @ref rPath is not a group.
+     *
+     * @param rPath     Path to be created.
+     */
     void AddPath(const std::string& rPath);
 
-    template<class TScalar>
-    void WriteAttribute(const std::string& rObjectPath, const std::string& rName, TScalar Value);
-
-    template<class TScalar>
-    void WriteAttribute(const std::string& rObjectPath, const std::string& rName, const Vector<TScalar>& rValue);
-
-    template<class TScalar>
-    void WriteAttribute(const std::string& rObjectPath, const std::string& rName, const Matrix<TScalar>& rValue);
-
-    void WriteAttribute(const std::string& rObjectPath, const std::string& rName, const std::string& rValue);
-
-    void WriteAttribute(const std::string& rObjectPath, const std::string& rName, const array_1d<double, 3>& rValue);
-
-    /// Write a data set to the HDF5 file.
     /**
-     *  Performs collective write in MPI. The data is written blockwise according to
-     *  processor rank.
+     * @brief Write attributes to specified group or data set.
      *
-     * @param[out] rInfo Information about the written data set.
+     * @tparam TDataType            Data type of the attribute.
+     * @param rObjectPath           Group or dataset path.
+     * @param rName                 Attribute name.
+     * @param rValue                Atribute data.
      */
-    virtual void WriteDataSet(const std::string& rPath, const Vector<int>& rData, WriteInfo& rInfo);
-
-    virtual void WriteDataSet(const std::string& rPath, const Vector<double>& rData, WriteInfo& rInfo);
-
-    virtual void WriteDataSet(const std::string& rPath, const Vector<array_1d<double, 3>>& rData, WriteInfo& rInfo);
+    template<class TDataType>
+    void WriteAttribute(
+        const std::string& rObjectPath,
+        const std::string& rName,
+        const TDataType& rValue);
 
-    virtual void WriteDataSet(const std::string& rPath, const Matrix<int>& rData, WriteInfo& rInfo);
+    /**
+     * @brief Write attributes in Parameters object to dataset or group.
```

Comment:
Here's your chance to use `@ref`. `Parameters` is a class that has its own entry in the docs.

```suggestion
     * @brief Write attributes provided in a @ref Parameters object to a dataset or group.
```

Reply:
done :)